code/WorkInProgress/MechanicMadness.dm

```diff
@@ -977,6 +977,38 @@ var/list/mechanics_telepads = new/list()
 		icon_state = "[under_floor ? "u":""]comp_accel"
 		return
 
+/obj/item/mechanics/zapper
+	name = "Tesla Coil"
+	desc = ""
+	icon_state = "comp_zap"
+	cooldown_time = 1 SECONDS
+	var/zap_power = 2
+	var/zap_radius = 0
+
+	New()
+		..()
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"zap", "eleczap")
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Power","setPower")
+
+	proc/eleczap(var/datum/mechanicsMessage/input)
+		if(level == 2) return
+		LIGHT_UP_HOUSING
+		elecflash(src.loc, 0, power = zap_power, exclude_center = 0)
+		return
+		
+	proc/setPower(obj/item/W as obj, mob/user as mob)
+		var/inp = input(user,"Please enter Power(1 - 6):","Power setting", zap_power) as num
+		if(!in_range(src, user) || user.stat)
```

Comment:
DIST_CHECK , also don't use raw .stat comparisons, use isalive(x) or whatever, you also want to probably do whatever that wrapper proc for paralyzed,stunned,downed, etc. is

Reply:
I renamed DIST_CHECK to IN_RANGE some time ago.

Reply:
DIST_CHECK was a confusing name and people often accidentally used it instead of `get_dist(a, b) >= X` because the name doesn't immediately imply <=.

Reply:
Zewaka marked this as resolved but I assume he did so by mistake because these are still issues